### ISSUE TITLE
Couldn't install tensotflow==2.1.0

### ISSUE THREAD

**Issue by zueve** [python-poetry/poetry#1962]
- [*] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) Poetry version.
- [*] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [*] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: Docker image python:3.6.9
- **Poetry version**: 1.0.2

## Issue

I have not way to install latest tensorflow with poetry==1.0.2. If try to install I have this trace:

```
[EnvCommandError]
Command ['/home/zenv/.cache/pypoetry/virtualenvs/1-ZUkQNI8o-py3.6/bin/pip', 'install', '--no-deps', 'tensorflow==2.1.0'] errored with the following return code 1, and output: 
Collecting tensorflow==2.1.0
  Could not find a version that satisfies the requirement tensorflow==2.1.0 (from versions: 0.12.1, 1.0.0, 1.0.1, 1.1.0rc0, 1.1.0rc1, 1.1.0rc2, 1.1.0, 1.2.0rc0, 1.2.0rc1, 1.2.0rc2, 1.2.0, 1.2.1, 1.3.0rc0, 1.3.0rc1, 1.3.0rc2, 1.3.0, 1.4.0rc0, 1.4.0rc1, 1.4.0, 1.4.1, 1.5.0rc0, 1.5.0rc1, 1.5.0, 1.5.1, 1.6.0rc0, 1.6.0rc1, 1.6.0, 1.7.0rc0, 1.7.0rc1, 1.7.0, 1.7.1, 1.8.0rc0, 1.8.0rc1, 1.8.0, 1.9.0rc0, 1.9.0rc1, 1.9.0rc2, 1.9.0, 1.10.0rc0, 1.10.0rc1, 1.10.0, 1.10.1, 1.11.0rc0, 1.11.0rc1, 1.11.0rc2, 1.11.0, 1.12.0rc0, 1.12.0rc1, 1.12.0rc2, 1.12.0, 1.12.2, 1.12.3, 1.13.0rc0, 1.13.0rc1, 1.13.0rc2, 1.13.1, 1.13.2, 1.14.0rc0, 1.14.0rc1, 1.14.0, 2.0.0a0, 2.0.0b0, 2.0.0b1)
No matching distribution found for tensorflow==2.1.0
You are using pip version 18.1, however version 20.0.2 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.

```

**Comment by Senpos** [comment]
[Official "Install" page](https://www.tensorflow.org/install/pip) says, that you are [required ](https://www.tensorflow.org/install/pip#system-requirements)to have `pip>=19` to install `tensorflow`:
```
System requirements
pip 19.0 or later (requires manylinux2010 support)
```

But you have `pip18.1`. I'd try running `pip install --upgrade pip` as the error message recommends before you install this package.

**Comment by Senpos** [comment]
>OS version and name: Docker image python:3.6.9

I am just curious, is there any specific reaason for you to use Python 3.6?

**Comment by zueve** [comment]
> I am just curious, is there any specific reaason for you to use Python 3.6

It's require for another libs, and offisal docker tf image use py36

> But you have pip18.1. I'd try running pip install --upgrade pip as the error message recommends before you install this package.

You are right! I fixed it after run
> poetry run pip install -U pip

But it's strange because before installl poetry i upgrade system pip to  20.0.2, but poetry use old version

**Comment by legel** [comment]
Thanks to @zueve for this update to pip that solves the problem:

`poetry run pip install -U pip`

giving the following:

> Collecting pip
>   Using cached https://files.pythonhosted.org/packages/54/0c/d01aa759fdc501a58f431eb594a17495f15b88da142ce14b5845662c13f3/pip-20.0.2-py2.py3-none-any.whl
> Installing collected packages: pip
>   Found existing installation: pip 9.0.1
>     Uninstalling pip-9.0.1:
>       Successfully uninstalled pip-9.0.1
> Successfully installed pip-20.0.2

**Comment by mbelang** [comment]
I did update pip and it is not working at all.

**Comment by zhenyulin** [comment]
still got the problem even after updated pip as well

**Comment by zhenyulin** [comment]
fixed my problem by switching from python 3.8 to 3.6

**Comment by stephsamson** [comment]
On Tensorflow's [install page](https://www.tensorflow.org/install/pip) they require a Python version that is >= 3.5, <3.8 which is why 3.6 worked but 3.8 didn't.

The page also specifies that one needs pip 19.0 or later.

TLDR to correctly install Tensorflow with poetry:

- Use a Python version that is from 3.5 to 3.7
- Use pip version 19.0 or later

I will be closing this issue now since @zueve's issue seems to have been resolved on their end. For any other questions regarding this topic, please ask under our [Stack Overflow tag](https://stackoverflow.com/tags/python-poetry/info) or join our [Discord](https://discordapp.com/invite/awxPgve) server.
